src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2394,11 +2394,17 @@ static int __icap_xclbin_download(struct icap *icap, struct axlf *xclbin)
 			goto out;
 		}
 	}
-
-
-	err = icap_download_bitstream(icap, xclbin);
-	if (err)
-		goto out;
+// dont need to program the bitstream for flat shell
+	if(xclbin->m_header.m_mode != XCLBIN_FLAT)
+	{
+    err = icap_download_bitstream(icap, xclbin);
+    if (err)
+      goto out;
+  }
+  else
+  {
+		ICAP_INFO(icap, "xclbin is generated for flat shell, dont need to load bitstream ");
+	}
```

Comment:
please follow driver coding style. e.g. Use tab ident and  put '{' to the same line with 'if'

Reply:
corrected as per coding style